src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -171,6 +170,28 @@ trait MarathonConf
     true
   }
 
+  private[this] def deriveDefaultAcceptedResourceRolesDefaultBehavior: Option[AcceptedResourceRolesDefaultBehavior] = {
+    defaultAcceptedResourceRoles.toOption match {
+      case Some(set) if set == Set(ResourceRole.Unreserved) => Some(AcceptedResourceRolesDefaultBehavior.Unreserved)
+      case Some(set) if set == Set(mesosRole()) => Some(AcceptedResourceRolesDefaultBehavior.Reserved)
+      case _ => Some(AcceptedResourceRolesDefaultBehavior.Any)
+    }
+  }
+
+  def defaultAcceptedResourceRolesSet(serviceRole: Role): Set[String] = {
+    acceptedResourceRolesDefaultBehavior() match {
+      case AcceptedResourceRolesDefaultBehavior.Any => Set(serviceRole, ResourceRole.Unreserved)
+      case AcceptedResourceRolesDefaultBehavior.Unreserved => Set(ResourceRole.Unreserved)
+      case AcceptedResourceRolesDefaultBehavior.Reserved => Set(serviceRole)
+    }
+  }
+
+  lazy val acceptedResourceRolesDefaultBehavior = opt[AcceptedResourceRolesDefaultBehavior](
```

Comment:
Should we validate that only one of these (`accepted_resource_roles_default_behavior and default_accepted_resource_roles`) parameters is used?

Reply:
Yeah, we should. I'll follow it up.